### PR TITLE
[BM Hunter] Azerite Trait: Dance of Death

### PR DESCRIPTION
**Pull request by hpabst**
- Adds Azerite trait [Dance of Death](http://www.wowhead.com/spell=274441/dance-of-death). Towards issue #2037 .

image

### CODE REVIEW

File: src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        icon={<SpellIcon id={SPELLS.DANCE_OF_DEATH.id} />}
+        value={(
+          <React.Fragment>
+            {formatNumber(this.avgAgility)} Agility <br />
+            {formatNumber(this.numProcs)} Procs
+          </React.Fragment>
+        )}
+        label={"Dance of Death"}
+        tooltip={`Dance of Deasth granted <b>${this.agility}</b> agility for <b>${formatPercentage(this.uptime)}%</b> of the fight.`}
+      />
```
Comment (Pewtro) [inline src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js]: spelling in the tooltip with Dance of Death

File: src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    return this.uptime * this.agility;
+  }
+
+  get numProcs(){
+    return this.selectedCombatant.getBuffTriggerCount(SPELLS.DANCE_OF_DEATH_BUFF.id);
+  }
+
+  statistic(){
+    return (
+      <StatisticBox
```
Comment (MartijnHols) [inline src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js]: Please use the (new) TraitStatisticBox for traits.

File: src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        icon={<SpellIcon id={SPELLS.DANCE_OF_DEATH.id} />}
+        value={(
+          <React.Fragment>
+            {formatNumber(this.avgAgility)} Agility <br />
+            {formatNumber(this.numProcs)} Procs
```
Comment (MartijnHols) [inline src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js]: How about showing the uptime? I think that could be more useful than procs.

File: src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      <TraitStatisticBox
+        position={STATISTIC_ORDER.OPTIONAL()}
+        trait={SPELLS.DANCE_OF_DEATH.id}
+        value={(
+          <React.Fragment>
+            {formatNumber(this.avgAgility)} Average Agility <br />
+            {formatPercentage(this.uptime)}% Uptime
+          </React.Fragment>
+        )}
+        label={"Dance of Death"}
```
Comment (MartijnHols) [inline src/Parser/Hunter/BeastMastery/Modules/Spells/AzeriteTraits/DanceOfDeath.js]: Please omit the label so the TraitStatisticBox shows a colored link.

**Pull request merged**